internal/injector/typed/parser.go

```diff
@@ -0,0 +1,403 @@
+//   - Slice types: "[]string", "[]*User"
+//   - Array types: "[10]string", "[0xFF]byte"
+//   - Map types: "map[string]int", "map[string]*User"
+//
+// Not yet supported (will return an error):
+//   - Channel types: "chan int", "<-chan int", "chan<- int"
+//   - Function types: "func()", "func(int) string"
+//   - Interface types: "interface{}", "interface{ Method() }"
+//   - Struct types: "struct{}", "struct{ Name string }"
+//   - Generic types: "List[T]", "Map[K, V]"
+//
+// TODO: Add support for these types if needed in the future.
+func (p *typeParser) parseTypeExpr() (Type, error) {
+	p.skipWhitespace()
+
+	if p.pos >= len(p.input) {
+		return nil, errors.New("unexpected end of input")
+	}
+
+	// Check for pointer
+	if p.input[p.pos] == '*' {
+		p.pos++
+		p.skipWhitespace()
+
+		// Check for invalid double pointer with space (e.g., "* *string")
+		if p.pos < len(p.input) && p.input[p.pos] == '*' {
+			return nil, errors.New("invalid pointer syntax: space between * operators")
+		}
+
+		elem, err := p.parseTypeExpr()
+		if err != nil {
+			return nil, err
+		}
+		return &PointerType{Elem: elem}, nil
+	}
+
+	// Check for slice or array
+	if p.input[p.pos] == '[' {
+		return p.parseSliceOrArray()
+	}
+
+	// Check for map
+	if p.consumeKeyword("map") {
+		return p.parseMap()
+	}
+
+	// TODO: Add support for channels (check for "chan" keyword)
+	// TODO: Add support for functions (check for "func" keyword)
+	// TODO: Add support for interfaces (check for "interface" keyword)
+	// TODO: Add support for structs (check for "struct" keyword)
+
+	// Otherwise, it must be a named type
+	return p.parseNamedType()
+}
+
+// parseSliceOrArray parses slice or array types starting after the '['.
+func (p *typeParser) parseSliceOrArray() (Type, error) {
+	p.pos++ // consume '['
+	p.skipWhitespace()
+
+	// Check if it's a slice (empty brackets)
+	if p.pos < len(p.input) && p.input[p.pos] == ']' {
+		p.pos++ // consume ']'
+		elem, err := p.parseTypeExpr()
+		if err != nil {
+			return nil, err
+		}
+		return &SliceType{Elem: elem}, nil
+	}
+
+	// It's an array, parse the size
+	size, err := p.parseArraySize()
+	if err != nil {
+		return nil, err
+	}
+
+	p.skipWhitespace()
+	if p.pos >= len(p.input) || p.input[p.pos] != ']' {
+		return nil, errors.New("expected ']' after array size")
+	}
+	p.pos++ // consume ']'
+
+	elem, err := p.parseTypeExpr()
+	if err != nil {
+		return nil, err
+	}
+
+	return &ArrayType{Size: size, Elem: elem}, nil
+}
+
+// parseArraySize parses an integer literal for array size.
+func (p *typeParser) parseArraySize() (int, error) {
+	start := p.pos
+
+	// Handle hex (0x...) or octal (0...) prefixes
+	if p.pos < len(p.input) && p.input[p.pos] == '0' {
+		p.pos++
+		if p.pos < len(p.input) && (p.input[p.pos] == 'x' || p.input[p.pos] == 'X') {
+			p.pos++
+			// Parse hex digits
+			if !p.consumeHexDigits() {
+				return 0, errors.New("invalid hex number")
+			}
+		} else {
+			// Parse octal digits
+			p.consumeOctalDigits()
```

Comment:
There might be an `o` in the way here, as only dangerous sociopaths use the leading 0 alone as an octal marker...